lean4/src/putnam_2023_b3.lean

```diff
@@ -0,0 +1,18 @@
+import Mathlib
+
+open ProbabilityTheory Set
+
+noncomputable abbrev putnam_2023_b3_solution : ℕ → ℝ := sorry
+-- (fun n : ℕ ↦ (2 * (n : ℝ) + 2) / 3)
+
+theorem putnam_2023_b3
+    (IsZigZag : {k : ℕ} → (Fin k → ℝ) → Prop)
+    (IsZigZag_def : ∀ (k : ℕ) (hk : NeZero k) (y : Fin k → ℝ),
```

Comment:
Maybe
```suggestion
    (IsZigZag_def : ∀ (k : ℕ) [NeZero k] (y : Fin k → ℝ),
```

Reply:
I remember considering this but I can't remember my reservations so I've adjusted and (force) pushed.